shell/platform/android/io/flutter/plugin/platform/PlatformView.java

```diff
@@ -16,13 +17,46 @@
      */
     View getView();
 
+    /**
+     * Called by the {@link FlutterEngine} that owns this {@code PlatformView} when
+     * the Android {@link View} responsible for rendering a Flutter UI is associated
+     * with the {@link FlutterEngine}.
+     *
+     * <p>This means that our associated {@link FlutterEngine} can now render a UI and
+     * interact with the user.
+     *
+     * <p>Some platform views may have unusual dependencies on the {@link View} that
+     * renders Flutter UIs, such as unique keyboard interactions. That {@link View}
+     * is provided here for those purposes. Use of this {@link View} should be avoided
+     * if it is not absolutely necessary, because depending on this {@link View} will
+     * tend to make platform view code more brittle to future changes.
+     */
+    default void onFlutterViewAttached(@NonNull View flutterView) {}
```

Comment:
@mklim what's the latest with the desugarer issue?
(before we're doubling down on default interface implementations)

Reply:
@blasten has found out that if you include the engine as an API dependency for plugins it fixes the problem, since then the desugaring doesn't strip out the "unneeded" symbols. The way we include the embedding is strange so it's hard for us to call if that's actually semantically correct or not, but I think it makes sense. All Flutter dependencies should be using the same version of the embedding, so exposing it at compile time via the API dependency shouldn't be a problem. So we're probably going to move forward with that fix.

I initially found a fallback where we can force the old dexing/desugaring pipeline with a flag, but I prefer @blasten's approach because it seems more future proof.